nucypher-core-python/src/lib.rs

```diff
@@ -573,20 +573,20 @@ pub struct ReencryptionResponse {
 #[pymethods]
 impl ReencryptionResponse {
     #[new]
-    pub fn new(signer: &Signer, capsules: Vec<Capsule>, vcfrags: Vec<VerifiedCapsuleFrag>) -> Self {
-        let capsules_backend = capsules
+    pub fn new(signer: &Signer, capsules_and_vcfrags: Vec<(Capsule, VerifiedCapsuleFrag)>) -> Self {
+        let (capsules_backend, vcfrags_backend): (Vec<_>, Vec<_>) = capsules_and_vcfrags
             .into_iter()
-            .map(umbral_pre::Capsule::from)
-            .collect::<Vec<_>>();
-        let vcfrags_backend = vcfrags
-            .into_iter()
-            .map(umbral_pre::VerifiedCapsuleFrag::from)
-            .collect::<Vec<_>>();
+            .map(|(capsule, vcfrag)| {
+                (
+                    umbral_pre::Capsule::from(capsule),
+                    umbral_pre::VerifiedCapsuleFrag::from(vcfrag),
+                )
+            })
+            .unzip();
```

Comment:
This is a lot for one line, cognitively speaking... so, we're basically mapping the objects to different types, and the unzipping the two sides of the iterable into two new locals, `capsules_backend` and `vcfrags_backend`?